Add link to `kubectl convert` (#40737)
kubernetes/website: content/en/docs/reference/using-api/deprecation-guide.md

Reviewer: We should reword “Migrate to non-deprecated APIs” to refer to `kubectl convert`, rather than `kubectl-convert`.

The current wording was put in as a quick rewrite pending the removal of the built-in subcommand.

Author: good point, made the reword here and used the "note" notation instead of obs that found in another place of the docs, should be more clean this way:

![image](https://user-images.githubusercontent.com/16529503/233292316-368feb71-0df4-4fde-aba6-3b8052031786.png)

* Update custom integrations and controllers to call the non-deprecated APIs
* Change YAML files to reference the non-deprecated APIs

  You can use the `kubectl convert` command to automatically convert an existing object:

  `kubectl convert -f <file> --output-version <group>/<version>`.

  For example, to convert an older Deployment to `apps/v1`, you can run:

  `kubectl convert -f ./my-deployment.yaml --output-version apps/v1`

>   Note that this may use non-ideal default values. To learn more about a specific

Reviewer: ```suggestion
  This conversion may use non-ideal default values. To learn more about a specific
```

  resource, check the Kubernetes [API reference](/docs/reference/kubernetes-api/).
  
  {{< note >}}

>     The `kubectl convert` command is not installed by default as before ([deprecation notes](https://github.com/kubernetes/kubectl/issues/725)).
>     
>     To learn how to set up `kubectl convert` on your computer, visit the follow links:
>     [Linux](https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/#install-kubectl-convert-plugin)
>     [MacOS](https://kubernetes.io/docs/tasks/tools/install-kubectl-macos/#install-kubectl-convert-plugin)
>     [Windows](https://kubernetes.io/docs/tasks/tools/install-kubectl-windows/#install-kubectl-convert-plugin)

Reviewer: ```suggestion
  The `kubectl convert` tool is not installed by default, although
  in fact it once was part of `kubectl` itself. For more details, you can read the
  [deprecation and removal issue](https://github.com/kubernetes/kubectl/issues/725)
  for the built-in subcommand).
  
  To learn how to set up `kubectl convert` on your computer, visit the page that is right for your 
  operating system:
  [Linux](https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/#install-kubectl-convert-plugin),
  [macOS](https://kubernetes.io/docs/tasks/tools/install-kubectl-macos/#install-kubectl-convert-plugin), or
  [Windows](https://kubernetes.io/docs/tasks/tools/install-kubectl-windows/#install-kubectl-convert-plugin).
```

Author: Thanks, didn't knew there was a page where i could see the preview :)

Seems to be okay now:

![image](https://user-images.githubusercontent.com/16529503/234883514-fdac971f-f717-4740-bf8d-270c2eec6c3a.png)

* Update custom integrations and controllers to call the non-deprecated APIs
* Change YAML files to reference the non-deprecated APIs

  You can use the `kubectl convert` command to automatically convert an existing object:

  `kubectl convert -f <file> --output-version <group>/<version>`.

  For example, to convert an older Deployment to `apps/v1`, you can run:

  `kubectl convert -f ./my-deployment.yaml --output-version apps/v1`

  This conversion may use non-ideal default values. To learn more about a specific
  resource, check the Kubernetes [API reference](/docs/reference/kubernetes-api/).
  
  {{< note >}}
  The `kubectl convert` tool is not installed by default, although
  in fact it once was part of `kubectl` itself. For more details, you can read the
  [deprecation and removal issue](https://github.com/kubernetes/kubectl/issues/725)
  for the built-in subcommand.
  
  To learn how to set up `kubectl convert` on your computer, visit the page that is right for your 
  operating system:

>   [Linux](https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/#install-kubectl-convert-plugin),

Reviewer 2: ```suggestion
  [Linux](/docs/tasks/tools/install-kubectl-linux/#install-kubectl-convert-plugin),
```

>   [macOS](https://kubernetes.io/docs/tasks/tools/install-kubectl-macos/#install-kubectl-convert-plugin), or

Reviewer 2: ```suggestion
  [macOS](/docs/tasks/tools/install-kubectl-macos/#install-kubectl-convert-plugin), or
```

>   [Windows](https://kubernetes.io/docs/tasks/tools/install-kubectl-windows/#install-kubectl-convert-plugin).

Reviewer 2: ```suggestion
  [Windows](/docs/tasks/tools/install-kubectl-windows/#install-kubectl-convert-plugin).
```

  Note that this may use non-ideal default values. To learn more about a specific
  resource, check the Kubernetes [API reference](/docs/reference/kubernetes-api/).
  

>   **OBS**: the `kubectl convert` command is not installed by default ([deprecation notes](https://github.com/kubernetes/kubectl/issues/725)) but needs to be installed as plugin installed separately. For more info, see how to install kubectl convert plugin on [Linux](https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/#install-kubectl-convert-plugin), [MacOS](https://kubernetes.io/docs/tasks/tools/install-kubectl-macos/#install-kubectl-convert-plugin) and [Windows](https://kubernetes.io/docs/tasks/tools/install-kubectl-windows/#install-kubectl-convert-plugin).

Reviewer: Hi. What does OBS mean?

>   **OBS**: the `kubectl convert` command is not installed by default ([deprecation notes](https://github.com/kubernetes/kubectl/issues/725)) but needs to be installed as plugin installed separately. For more info, see how to install kubectl convert plugin on [Linux](https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/#install-kubectl-convert-plugin), [MacOS](https://kubernetes.io/docs/tasks/tools/install-kubectl-macos/#install-kubectl-convert-plugin) and [Windows](https://kubernetes.io/docs/tasks/tools/install-kubectl-windows/#install-kubectl-convert-plugin).

Author: We can remove the OBS, it meant just as "observation". My problem with this approach is that it not leads directly to the section that explains the install process, but the user needs to click on the link in the section and roll all the way down to the `install-kubectl-convert-plugin` section.
